### PR TITLE
PLAT-4876: Unit test for symphony-utility module

### PR DESCRIPTION
**Pull request by ardsilva**

### CODE REVIEW

File: helpdesk-utility/src/main/java/org/symphonyoss/symphony/bots/utility/message/EntityBuilder.java [hunk 1]
```diff
@@ -11,13 +11,13 @@
  * <p>
  * Created by rsanchez on 12/01/17.
  */
-public class EntityBuilder {
+public class  EntityBuilder {
```
Comment (robson-sanchez) [inline helpdesk-utility/src/main/java/org/symphonyoss/symphony/bots/utility/message/EntityBuilder.java]: Please remove the white space.

File: helpdesk-utility/src/main/java/org/symphonyoss/symphony/bots/utility/message/EntityBuilder.java [hunk 2]
```diff
@@ -11,13 +11,13 @@
...
 
   private static final ObjectMapper MAPPER = new ObjectMapper();
 
   private Map<String, Object> content = new LinkedHashMap<>();
 
-  private EntityBuilder() {
+  protected EntityBuilder() {
```
Comment (robson-sanchez) [inline helpdesk-utility/src/main/java/org/symphonyoss/symphony/bots/utility/message/EntityBuilder.java]: You shouldn't update the class scope. Please use the 'createEntity' static method to instantiate this class.

File: helpdesk-utility/src/main/java/org/symphonyoss/symphony/bots/utility/message/SymMessageBuilder.java [hunk 3]
```diff
@@ -19,7 +19,7 @@
 
   private List<SymAttachmentInfo> attachments = new ArrayList<>();
 
-  private SymMessageBuilder(String message) {
+  protected SymMessageBuilder(String message) {
```
Comment (robson-sanchez) [inline helpdesk-utility/src/main/java/org/symphonyoss/symphony/bots/utility/message/SymMessageBuilder.java]: You shouldn't update the class scope. Please use the 'message' static method to instantiate this class.

File: helpdesk-utility/src/test/java/org/symphonyoss/symphony/bots/utility/config/ServiceInfoTest.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+  private ServiceInfo serviceInfo;
+
+  @Before
+  public void setUp() throws Exception {
+    serviceInfo = new ServiceInfo();
+  }
+
+  @Test
+  public void getHost() {
+    serviceInfo.getHost();
```
Comment (robson-sanchez) [inline helpdesk-utility/src/test/java/org/symphonyoss/symphony/bots/utility/config/ServiceInfoTest.java]: This statement is not required.

File: helpdesk-utility/src/test/java/org/symphonyoss/symphony/bots/utility/config/ServiceInfoTest.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+  @Test
+  public void setHost() {
+    serviceInfo.setHost(host);
+    assertEquals(host,serviceInfo.getHost());
+  }
+
+  @Test
+  public void getPort() {
+    serviceInfo.getPort();
```
Comment (robson-sanchez) [inline helpdesk-utility/src/test/java/org/symphonyoss/symphony/bots/utility/config/ServiceInfoTest.java]: This statement is not required.

File: helpdesk-utility/src/test/java/org/symphonyoss/symphony/bots/utility/config/ServiceInfoTest.java [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ServiceInfoTest {
+
+  private final String host = "helpdesk-test";
```
Comment (robson-sanchez) [inline helpdesk-utility/src/test/java/org/symphonyoss/symphony/bots/utility/config/ServiceInfoTest.java]: It should be a constant field.

```
private static final String HOST = "helpdesk-test.symphony.com";
```

File: helpdesk-utility/src/test/java/org/symphonyoss/symphony/bots/utility/config/ServiceInfoTest.java (hunk 5 above)
Comment (robson-sanchez) [inline helpdesk-utility/src/test/java/org/symphonyoss/symphony/bots/utility/config/ServiceInfoTest.java]: This statement is not required.

File: helpdesk-utility/src/test/java/org/symphonyoss/symphony/bots/utility/config/ServiceInfoTest.java [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+
+  private final Integer port = new Integer("8080");
```
Comment (robson-sanchez) [inline helpdesk-utility/src/test/java/org/symphonyoss/symphony/bots/utility/config/ServiceInfoTest.java]: It should be a constant field.

```
private static final Integer PORT = 8080;
```

File: helpdesk-utility/src/test/java/org/symphonyoss/symphony/bots/utility/message/EntityBuilderTest.java [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+  public void setUp() throws Exception {
+    entityBuilder = new EntityBuilder();
+  }
+
+  @Test
+  public void createEntity() {
+    EntityBuilder entity = entityBuilder.createEntity();
+    Map<String, Object> object = entity.toObject();
+    assertTrue(object.isEmpty());
+    assertEquals(0, object.size());
```
Comment (robson-sanchez) [inline helpdesk-utility/src/test/java/org/symphonyoss/symphony/bots/utility/message/EntityBuilderTest.java]: This statement is not required.

File: helpdesk-utility/src/test/java/org/symphonyoss/symphony/bots/utility/message/EntityBuilderTest.java [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+
+
+  private static final String FIELD = "FIELD";
+  private static final String TYPE = "TYPE";
+  private static final String VERSION = "VERSION";
+
+  private EntityBuilder entityBuilder;
+
+  @Before
+  public void setUp() throws Exception {
```
Comment (robson-sanchez) [inline helpdesk-utility/src/test/java/org/symphonyoss/symphony/bots/utility/message/EntityBuilderTest.java]: Please remove this method.

File: helpdesk-utility/src/test/java/org/symphonyoss/symphony/bots/utility/message/EntityBuilderTest.java [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+
+@RunWith(MockitoJUnitRunner.class)
+public class EntityBuilderTest {
+
+
+  private static final String FIELD = "FIELD";
+  private static final String TYPE = "TYPE";
+  private static final String VERSION = "VERSION";
+
+  private EntityBuilder entityBuilder;
```
Comment (robson-sanchez) [inline helpdesk-utility/src/test/java/org/symphonyoss/symphony/bots/utility/message/EntityBuilderTest.java]: This field is not required.

File: helpdesk-utility/src/test/java/org/symphonyoss/symphony/bots/utility/message/EntityBuilderTest.java [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+    Map<String, Object> object = entity.toObject();
+    assertTrue(object.isEmpty());
+    assertEquals(0, object.size());
+  }
+
+  @Test
+  public void createEntityWithValues() {
+    EntityBuilder builder = entityBuilder.createEntity(TYPE,VERSION);
+    Map<String, Object> object = builder.toObject();
+    assertNotNull(object.get("type"));
```
Comment (robson-sanchez) [inline helpdesk-utility/src/test/java/org/symphonyoss/symphony/bots/utility/message/EntityBuilderTest.java]: Please use assertEquals insteadof assertNotNull.

File: helpdesk-utility/src/test/java/org/symphonyoss/symphony/bots/utility/message/EntityBuilderTest.java [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+    assertNotNull(object.get("version"));
+    assertEquals(2, object.size());
+  }
+
+  @Test
+  public void addField() {
+    EntityBuilder builder = entityBuilder.addField(FIELD, FIELD);
+    Map<String, Object> object = builder.toObject();
+    assertNotNull(object.get(FIELD));
```
Comment (robson-sanchez) [inline helpdesk-utility/src/test/java/org/symphonyoss/symphony/bots/utility/message/EntityBuilderTest.java]: Please use assertEquals insteadof assertNotNull.

File: helpdesk-utility/src/test/java/org/symphonyoss/symphony/bots/utility/message/SymMessageBuilderTest.java [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+  private SymMessageBuilder symMessageBuilder;
+
+  @Before
+  public void setUp() throws Exception {
+    symMessageBuilder = new SymMessageBuilder(NEW_MESSAGE);
+  }
+
+  @Test
+  public void message() {
+    symMessageBuilder.message(NEW_MESSAGE);
```
Comment (robson-sanchez) [inline helpdesk-utility/src/test/java/org/symphonyoss/symphony/bots/utility/message/SymMessageBuilderTest.java]: There are no assertions here :(

File: helpdesk-utility/src/test/java/org/symphonyoss/symphony/bots/utility/validation/SymphonyValidationUtilTest.java [hunk 14]
```diff
@@ -0,0 +1,108 @@
...
+    assertNotNull(symUser);
+    assertEquals(getSymUser(), symUser);
+  }
+
+  @Test
+  public void validateUserIdThrowsException() throws UsersClientException {
+    doReturn(usersClient).when(symphonyClient).getUsersClient();
+    doThrow(UsersClientException.class).when(usersClient).getUserFromId(INVALID_ID);
+    try {
+      symphonyValidationUtil.validateUserId(INVALID_ID);
```
Comment (robson-sanchez) [inline helpdesk-utility/src/test/java/org/symphonyoss/symphony/bots/utility/validation/SymphonyValidationUtilTest.java]: Please change this to USER_ID.

**Review by robson-sanchez — CHANGES_REQUESTED** [review]

File: helpdesk-utility/src/main/java/org/symphonyoss/symphony/bots/utility/message/EntityBuilder.java (hunk 1 above)
Comment (ardsilva) [reply helpdesk-utility/src/main/java/org/symphonyoss/symphony/bots/utility/message/EntityBuilder.java]: ok

File: helpdesk-utility/src/main/java/org/symphonyoss/symphony/bots/utility/message/EntityBuilder.java (hunk 2 above)
Comment (ardsilva) [reply helpdesk-utility/src/main/java/org/symphonyoss/symphony/bots/utility/message/EntityBuilder.java]: ok

File: helpdesk-utility/src/main/java/org/symphonyoss/symphony/bots/utility/message/SymMessageBuilder.java (hunk 3 above)
Comment (ardsilva) [reply helpdesk-utility/src/main/java/org/symphonyoss/symphony/bots/utility/message/SymMessageBuilder.java]: ok

File: helpdesk-utility/src/test/java/org/symphonyoss/symphony/bots/utility/config/ServiceInfoTest.java (hunk 4 above)
Comment (ardsilva) [reply helpdesk-utility/src/test/java/org/symphonyoss/symphony/bots/utility/config/ServiceInfoTest.java]: ok

File: helpdesk-utility/src/test/java/org/symphonyoss/symphony/bots/utility/config/ServiceInfoTest.java (hunk 5 above)
Comment (ardsilva) [reply helpdesk-utility/src/test/java/org/symphonyoss/symphony/bots/utility/config/ServiceInfoTest.java]: ok

File: helpdesk-utility/src/test/java/org/symphonyoss/symphony/bots/utility/config/ServiceInfoTest.java (hunk 6 above)
Comment (ardsilva) [reply helpdesk-utility/src/test/java/org/symphonyoss/symphony/bots/utility/config/ServiceInfoTest.java]: ok

File: helpdesk-utility/src/test/java/org/symphonyoss/symphony/bots/utility/config/ServiceInfoTest.java (hunk 5 above)
Comment (ardsilva) [reply helpdesk-utility/src/test/java/org/symphonyoss/symphony/bots/utility/config/ServiceInfoTest.java]: ok

File: helpdesk-utility/src/test/java/org/symphonyoss/symphony/bots/utility/config/ServiceInfoTest.java (hunk 7 above)
Comment (ardsilva) [reply helpdesk-utility/src/test/java/org/symphonyoss/symphony/bots/utility/config/ServiceInfoTest.java]: ok

File: helpdesk-utility/src/test/java/org/symphonyoss/symphony/bots/utility/message/EntityBuilderTest.java (hunk 8 above)
Comment (ardsilva) [reply helpdesk-utility/src/test/java/org/symphonyoss/symphony/bots/utility/message/EntityBuilderTest.java]: ok

File: helpdesk-utility/src/test/java/org/symphonyoss/symphony/bots/utility/message/EntityBuilderTest.java (hunk 9 above)
Comment (ardsilva) [reply helpdesk-utility/src/test/java/org/symphonyoss/symphony/bots/utility/message/EntityBuilderTest.java]: ok

File: helpdesk-utility/src/test/java/org/symphonyoss/symphony/bots/utility/message/EntityBuilderTest.java (hunk 10 above)
Comment (ardsilva) [reply helpdesk-utility/src/test/java/org/symphonyoss/symphony/bots/utility/message/EntityBuilderTest.java]: ok

File: helpdesk-utility/src/test/java/org/symphonyoss/symphony/bots/utility/message/EntityBuilderTest.java (hunk 11 above)
Comment (ardsilva) [reply helpdesk-utility/src/test/java/org/symphonyoss/symphony/bots/utility/message/EntityBuilderTest.java]: ok

File: helpdesk-utility/src/test/java/org/symphonyoss/symphony/bots/utility/message/EntityBuilderTest.java (hunk 12 above)
Comment (ardsilva) [reply helpdesk-utility/src/test/java/org/symphonyoss/symphony/bots/utility/message/EntityBuilderTest.java]: ok

File: helpdesk-utility/src/test/java/org/symphonyoss/symphony/bots/utility/validation/SymphonyValidationUtilTest.java (hunk 14 above)
Comment (ardsilva) [reply helpdesk-utility/src/test/java/org/symphonyoss/symphony/bots/utility/validation/SymphonyValidationUtilTest.java]: ok

**Review by robson-sanchez — APPROVED** [review]

**Pull request merged**